src/topology/category/Compactum.lean

```diff
@@ -0,0 +1,467 @@
+    { suffices : (ultrafilter.map X.str GG).1 ≤ (X.incl x).1,
+      { ext1,
+        exact is_ultrafilter.unique (incl X x).2 (ultrafilter.map (str X) GG).2.1 this },
+      intros P hP,
+      apply filter.sets_of_superset _ (h1 (or.inl rfl)),
+      rintros x ⟨rfl⟩,
+      exact hP },
+    simp [←c1,c2] },
+  -- Finish...
+  intros T hT,
+  refine claim6 _ (finite_inter_mem (finite_inter_closure_has_finite_inter _) _ _),
+  intros t ht,
+  exact finite_inter_closure.basic (@hT t ht)
+end
+
+lemma le_nhds_of_str_eq {X : Compactum} (F : ultrafilter X) (x : X) :
+  X.str F = x → F.1 ≤ nhds x := λ h, le_nhds_iff.mpr (λ s hx hs, hs _ $ by rwa h)
+
+-- All the hard work above boils down to this t2_space instance.
+instance {X : Compactum} : t2_space X :=
+begin
+  rw t2_iff_ultrafilter,
+  intros _ _ _ hF hx hy,
+  replace hx := str_eq_of_le_nhds ⟨_,hF⟩ _ hx,
+  replace hy := str_eq_of_le_nhds ⟨_,hF⟩ _ hy,
+  cc,
+end
+
+/-- Any morphism of compacta is continuous. -/
+lemma continuous_of_hom {X Y : Compactum} (f : X ⟶ Y) : continuous f :=
+begin
+  rw continuous_iff_ultrafilter,
+  intros x _ h1 h2,
+  change (ultrafilter.map f ⟨_, h1⟩).1 ≤ _,
+  apply le_nhds_of_str_eq,
+  rw [←str_hom_commute, str_eq_of_le_nhds ⟨_, h1⟩ x h2],
+end
+
+/-- Given any compact Hausdorff space, we construct a Compactum. -/
+noncomputable def of_topological_space (X : Type*) [topological_space X]
+  [compact_space X] [t2_space X] : Compactum :=
+{ A := X,
+  a := ultrafilter.Lim,
+  unit' := by {ext x, exact Lim_eq (by finish [le_nhds_iff]) },
+  assoc' := begin
+    ext FF,
+    change ultrafilter (ultrafilter X) at FF,
+    set x := (ultrafilter.map ultrafilter.Lim FF).Lim with c1,
+    have c2 : ∀ (U : set X) (F : ultrafilter X), F.Lim ∈ U → is_open U → U ∈ F.1,
+    { intros U F h1 hU,
+      exact c1 ▸ is_open_iff_ultrafilter.mp hU _ h1 _ F.2 (is_ultrafilter.le_nhds_Lim _) },
+    have c3 : (ultrafilter.map ultrafilter.Lim FF).1 ≤ nhds x,
+    { rw le_nhds_iff,
+      intros U hx hU,
+      refine c2 _ _ (by rwa ← c1) hU },
+    have c4 : ∀ (U : set X), x ∈ U → is_open U → { G : ultrafilter X | U ∈ G.1 } ∈ FF.1,
+    { intros U hx hU,
+      suffices : ultrafilter.Lim ⁻¹' U ∈ FF.1,
+      { apply filter.sets_of_superset _ this,
+        intros P hP,
+        exact c2 U P hP hU},
+      exact @c3 U (mem_nhds_sets hU hx) },
+    apply Lim_eq,
+    rw le_nhds_iff,
+    exact c4,
+  end }
+
+/-- Any continuous map between Compacta is a morphism of compacta. -/
+def hom_of_continuous {X Y : Compactum} (f : X → Y) (cont : continuous f) : X ⟶ Y :=
+{ f := f,
+  h' := begin
+    rw continuous_iff_ultrafilter at cont,
+    ext (F : ultrafilter X),
+    specialize cont (X.str F) F.1 F.2 (le_nhds_of_str_eq F (X.str F) rfl),
+    have := str_eq_of_le_nhds (ultrafilter.map f F) _ cont,
+    simpa only [←this, types_comp_apply, of_type_functor_map],
+  end }
+
+end Compactum
+
+/-- The type of Compact Hausdorff topological spaces. -/
```

Comment:
This may be better in a less specialized file.

Reply:
Any suggestion on where this should go? In the file defining `compact_space` or in the file defining `t2_space`, or in a new file `topology/category/CompHaus.lean`?

Reply:
I moved this to a new file `topology/category/CompHaus.lean`.